virttest/utils_test/qemu/__init__.py

```diff
@@ -698,3 +699,309 @@ def get_all_memorys(self, vm):
+
+    """
+    Provide basic test for usb devices
+    """
+
+    def __init__(self, test, params, env):
+        self.params = params
+        self.test = test
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+
+    def check_usb_dev_monitor(self, usb_type):
+        """
+        check if the usb device exist in monitor
+        """
+        logging.info("Verify USB device in monitor.")
+        o = self.vm.monitor.info("usb")
+        devnums =[]
+        devs =[]
+        for line in o.splitlines():
+            if "testdev" in line:
+                for i in range(2):
+                    num = re.search("\d+(\.\d)?", line.split(",")[i]).group(0)
+                    devnums.append(num)
+                devs.append(devnums)
+        if devs:
+            return devs
+        else:
+            raise exceptions.TestFail("Could not find '%s' device, monitor "
+                                      "returns: \n%s" % (usb_type, o))
+
+    def check_usb_dev(self, usb_type):
+        """
+        check if the usb device exist in vm
+        """
+        if usb_type == 'usb-storage' or usb_type == 'usb-bot':
+            self.add_udisk()
+        timeout = float(self.params.get("login_timeout", 600))
+        session = self.vm.wait_for_login(timeout=timeout)
+        devs = self.check_usb_dev_monitor(usb_type)
+        if devs:
+            logging.info("The %s device can find in the monitor." % usb_type)
+        exist = self.check_usb_dev_guest(usb_type, session, devs)
+        if exist:
+            logging.info(" %s %s device was found in guest" % (exist, usb_type)
+        else:
+            raise exceptions.TestFail("Could not find '%s' in guest" % usb_type)
+
+    def check_usb_dev_guest(self, usb_type, session, devs):
+        """
+        check if the usb device exist in guest
+        """
+        logging.info("Verify %s device in guest." % usb_type)
+        cmd ="lsusb.py"
+        output = session.cmd_output(cmd)
+        num = 0
+        for devnum in devs:
+            for i in range(1,3):
+                key = "%s-%s" % (int(devnum[0][0])+i, devnum[1])
+                if key not in output:
+                    continue
+                else:
+                    num += 1
+        if num > 0:
+            return num
+        else:
+            return False
+
+    def create_img(self, image="usb", fmt="qcow2", size="200M"):
+
+        """
+        create image file for drive
+        """
+        self.params['images'] += " %s" % image
+        self.params["image_name_%s" % image] = "images/%s" % image
+        self.params["image_format_%s" % image] = fmt
+        self.params["image_size_%s" % image] = size
+        self.params["remove_image_%s" % image] = "yes"
+        self.params["create_image_%s" % image] = "yes"
+        env_process.process_images(env_process.preprocess_image,
+                                   self.test, self.params)
+
+    def drive_set(self, image, interface='none', fmt='qcow2',
```

Comment:
I think you just need to add usb_hub in your configuration, and set your usb_bot's "pci_bus= usb_hub.aid"